crates/fdev/src/build.rs

```diff
@@ -259,65 +274,66 @@ mod contract {
-                    .args(["install"])
-                    .current_dir(cwd)
-                    .stdout(Stdio::piped())
-                    .stderr(Stdio::piped())
-                    .spawn()
-                    .map_err(|e| {
-                        eprintln!("Error while installing npm packages: {e}");
-                        Error::CommandFailed("npm")
-                    })?;
-                pipe_std_streams(child)?;
-                let webpack = web_config
-                    .typescript
-                    .as_ref()
-                    .map(|c| c.webpack)
-                    .unwrap_or_default();
-                use std::io::IsTerminal;
-                if webpack {
-                    let cmd_args: &[&str] =
-                        if std::io::stdout().is_terminal() && std::io::stderr().is_terminal() {
-                            &["--color"]
-                        } else {
-                            &[]
-                        };
-                    let child = Command::new("webpack")
-                        .args(cmd_args)
+        if let Some(web_config) = &config.webapp {
+            println!("Bundling webapp contract state");
+            match &web_config.lang {
+                Some(SupportedWebLangs::Typescript) => {
+                    let child = Command::new(NPM_BUILD_COMMAND)
+                        .args(["install"])
                         .current_dir(cwd)
                         .stdout(Stdio::piped())
                         .stderr(Stdio::piped())
                         .spawn()
                         .map_err(|e| {
-                            eprintln!("Error while executing webpack command: {e}");
-                            Error::CommandFailed("tsc")
+                            eprintln!("Error while installing npm packages: {e}");
+                            Error::CommandFailed(NPM_BUILD_COMMAND)
                         })?;
                     pipe_std_streams(child)?;
-                    println!("Compiled input using webpack");
-                } else {
-                    let cmd_args: &[&str] =
-                        if std::io::stdout().is_terminal() && std::io::stderr().is_terminal() {
-                            &["--pretty"]
-                        } else {
-                            &[]
-                        };
-                    let child = Command::new("tsc")
-                        .args(cmd_args)
-                        .current_dir(cwd)
-                        .stdout(Stdio::piped())
-                        .stderr(Stdio::piped())
-                        .spawn()
-                        .map_err(|e| {
-                            eprintln!("Error while executing command tsc: {e}");
-                            Error::CommandFailed("tsc")
-                        })?;
-                    pipe_std_streams(child)?;
-                    println!("Compiled input using tsc");
+                    let webpack = web_config
```

Comment:
this is redundant if we are checking for webpack already above (line 277);
the else condition of 298 won't event be ever be executed here, so it needs to be moved to an else branch for line 277